src/hooks/form/types.ts

```diff
@@ -0,0 +1,104 @@
+// Core form types
+import { ComponentChild } from 'preact';
+
+export type FieldValue = unknown;
```

Comment:
Can this perhaps be narrowed down to: 
```suggestion
export type FieldValue = string | number | boolean;
```